pkg/v1/providers/yttcb/clusterbootstrap.yaml

```diff
@@ -0,0 +1,129 @@
+#@           data.values.ANTREA_ENDPOINTSLICE != True or
+#@           data.values.ANTREA_POLICY != True or
+#@           data.values.ANTREA_NODEPORTLOCAL != False or
+#@           data.values.ANTREA_TRACEFLOW != True or
+#@           data.values.ANTREA_EGRESS != False or
+#@           data.values.ANTREA_FLOWEXPORTER != False or
+#@           data.values.ANTREA_DISABLE_UDP_TUNNEL_OFFLOAD != False)
+#@ end
+
+#@ def is_vspherecsi_configs_given():
+#@   return (data.values.VSPHERE_INSECURE != True or
+#@           data.values.USE_TOPOLOGY_CATEGORIES != False)
+#@ end
+
+#@ def should_create_clusterbootstrap():
+#@   return data.values.CNI != "antrea" or is_antrea_configs_given() or is_vspherecsi_configs_given()
+#@ end
+
+#@ if should_create_clusterbootstrap():
+---
+apiVersion: run.tanzu.vmware.com/v1alpha3
+kind: ClusterBootstrap
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  namespace: #@ data.values.NAMESPACE
+  annotations:
+    tkg.tanzu.vmware.com/add-missing-fields-from-tkr: #@ "{}".format(get_tkr_version_from_tkr_name(data.values.KUBERNETES_RELEASE))
+spec:
+  cni:
+    #@ if data.values.CNI == "antrea":
+    refName: antrea*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: AntreaConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ elif data.values.CNI == "calico":
+    refName: calico*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: CalicoConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ end
+  kapp:
+    refName: kapp-controller*
+    valuesFrom:
+      providerRef:
+        apiGroup: run.tanzu.vmware.com
+        kind: KappControllerConfig
+        name: #@ data.values.CLUSTER_NAME
+  csi:
+    refName: vsphere*
+    valuesFrom:
+      providerRef:
+        apiGroup: csi.tanzu.vmware.com
+        kind: VSphereCSIConfig
+        name: #@ data.values.CLUSTER_NAME
+#@ end
+
+---
+#@ if is_antrea_configs_given():
+apiVersion: cni.tanzu.vmware.com/v1alpha1
+kind: AntreaConfig
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  namespace: #@ data.values.NAMESPACE
+spec:
+  antrea:
+    config:
+      #@ if data.values.NSXT_POD_ROUTING_ENABLED:
+      trafficEncapMode: "noEncap"
+      noSNAT: true
+      #@ else:
+      trafficEncapMode: #@ data.values.ANTREA_TRAFFIC_ENCAP_MODE
+      noSNAT: #@ data.values.ANTREA_NO_SNAT
+      #@ end
+      disableUdpTunnelOffload: #@ data.values.ANTREA_DISABLE_UDP_TUNNEL_OFFLOAD
+      featureGates:
+        #@ if data.values.NSXT_POD_ROUTING_ENABLED:
+        AntreaProxy: true
+        #@ else:
+        AntreaProxy: #@ data.values.ANTREA_PROXY
+        #@ end
+        EndpointSlice: #@ data.values.ANTREA_ENDPOINTSLICE
+        AntreaPolicy: #@ data.values.ANTREA_POLICY
+        NodePortLocal: #@ data.values.ANTREA_NODEPORTLOCAL
+        AntreaTraceflow: #@ data.values.ANTREA_TRACEFLOW
+        Egress: #@ data.values.ANTREA_EGRESS
+        FlowExporter: #@ data.values.ANTREA_FLOWEXPORTER
```

Comment:
Is this `ANTREA_*` configuration will be used for management-cluster only or the same will be used for workload cluster as well?

Because these `ANTREA_*` configuration might not be available with new ClusterClass based CCluster.yaml. cc @vuil

Reply:
Legacy configuration file used for workload cluster creation that requires Antrea customization should be leverage this overlay to produce the ClusterBootstrap configuration as well. So these data values are supported as long as the legacy config format is supported.

Reply:
@knabben - Your PR https://github.com/vmware-tanzu/tanzu-framework/pull/2353/files#diff-cab08af9fbfec8a91cfeefb2f2b30ed9b7dbe71809cc4f9a356e7afae1e1f45b is introducing 16 new variables for legacy antrea configuration. If that PR lands in 1.6, could you ensure that they are ported over here?

Reply:
yes, will port them over here after the PR merge

how does this will work with 1.2.3 legacy though?